### PR TITLE
Add full-lexer feature

### PR DESCRIPTION
**Pull request by youknowone**
Disabling full-lexer feature makes parsing `Lib/**/*.py` 4% faster.

Ruff will need to enable this feature to use `Tok::Comment`

cc @MichaReiser @charliermarsh

### CODE REVIEW

File: parser/src/python.lalrpop [hunk 1]
```diff
@@ -1743,6 +1743,6 @@ extern {
         name => token::Tok::Name { name: <String> },
         "\n" => token::Tok::Newline,
         ";" => token::Tok::Semi,
-        "#" => token::Tok::Comment(_),
+        // "#" => token::Tok::Comment(_),
```
Comment (DimitrisJim) [inline parser/src/python.lalrpop:1746]: hm, how does this work now?

Comment (youknowone) [reply parser/src/python.lalrpop:1746]: We already filter Comment before calling parser. So it was meaningless there for either feature flag on or off

File: parser/src/lexer.rs [hunk 2]
```diff
@@ -465,6 +466,20 @@ where
         }
     }
 
+    /// abandon a comment
```
Comment (DimitrisJim) [inline parser/src/lexer.rs:469]:
```suggestion
    /// Discard comment if full-lexer is not enabled.
```

File: parser/src/python.lalrpop (hunk 1 above)
Comment (DimitrisJim) [reply parser/src/python.lalrpop:1746]: ah. right!

**Review by DimitrisJim — APPROVED** [review]
I think this was mentioned in the past at some point. I think its a good idea to have a feature like this.

File: parser/src/lexer.rs (hunk 2 above)
Comment (youknowone) [reply parser/src/lexer.rs:469]: The function return type is different. Maybe this part can be `self.read_until_line_end()` later

**Pull request merged**